packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -724,10 +728,10 @@ export class IndexedDbPersistence implements Persistence {
     // Do all transactions as readwrite against all object stores, since we
     // are the only reader/writer.
     return this.simpleDb.runTransaction(
-      'readwrite',
+      mode == 'readonly' ? 'readonly' : 'readwrite',
```

Comment:
If you end up touching this PR before submitting (e.g. to remove the TODO), change this to `===`?  Else it's not really important.

Reply:
Done.